src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConditionalRestrictionCheck.java

```diff
@@ -0,0 +1,223 @@
+            "both");
+    private static final List<String> ACCESS_RESTRICTION_VALUE = List.of("yes", "no", "private",
+            "permissive", "destination", "delivery", "customers", "designated", "use_sidepath",
+            "dismount", "agricultural", "forestry", "discouraged", "official", "lane",
+            "share_busway", "opposite_share_busway");
+
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            "The conditional key \"{0}\" does not respect the \"<restriction-type>[:<transportation mode>][:<direction>]:conditional\" format",
+            "The conditional value \"{0}\" does not respect the format \"<restriction-value> @ <condition>[;<restriction-value> @ <condition>]\" ",
+            "The element with id {0,number,#} does not follow the conditional restriction pattern.");
+    public static final int TWO_PARTS = 2;
+    public static final int THREE_PARTS = 3;
+    public static final int FOUR_PARTS = 4;
+
+    public ConditionalRestrictionCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // all elements that have a tag containing ":conditional"
+        return object.getOsmTags().keySet().stream().anyMatch(key -> key.contains(CONDITIONAL));
+    }
+
+    /**
+     * Checks if the conditional restrictions respects the format
+     *
+     * @param object
+     *            the atlas object containing a conditional tag
+     * @return an optional {@link CheckFlag} object
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Set<String> instructions = new HashSet<>();
+
+        final List<String> conditionalKeys = object.getOsmTags().keySet().stream()
+                .filter(key -> key.contains(CONDITIONAL)).collect(Collectors.toList());
+        for (final String key : conditionalKeys)
+        {
+            if (!isKeyValid(key))
+            {
+                instructions.add(this.getLocalizedInstruction(0, key));
+            }
+            final String value = object.getOsmTags().get(key);
+            if (!isValueValid(value, key))
+            {
+                instructions.add(this.getLocalizedInstruction(1, value));
+            }
+        }
+        if (!instructions.isEmpty())
+        {
+            final CheckFlag flag = this.createFlag(object,
+                    this.getLocalizedInstruction(2, object.getOsmIdentifier()));
+            instructions.forEach(flag::addInstruction);
+            return Optional.of(flag);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    private boolean containsTransportationMode(final String key)
+    {
+        final String[] parts = key.split(":");
```

Comment:
The CommonConstants class cab be leveraged here: https://github.com/osmlab/atlas-checks/blob/d070ac474ff5751c12b98fde8fe4620a3d8ae131/src/main/java/org/openstreetmap/atlas/checks/constants/CommonConstants.java#L28